packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -363,6 +363,13 @@ async function responsiveSizes({ file, args = {} }) {
     options.sizes = `(max-width: ${options.maxWidth}px) 100vw, ${options.maxWidth}px`
   }
 
+  // Account for images with a high pixel density. We assume that these types
+  // of images are intended to be used as high resolution ("retina") images.
+  const { width, height, density } = await sharp(file.absolutePath).metadata()
```

Comment:
It'd be much more efficient to get the metadata from the previous sharp instance for the file. There's costs to making this call as it has to read the image off disk, do analysis, etc.

Reply:
Which previous sharp instance do you mean? AFAIK this is the first one. I could only try to pass this instance down to `gatsby-plugin-sharp`.